CRM/Member/Form/MembershipView.php

```diff
@@ -154,8 +154,21 @@ public function preProcess() {
     $this->assign('context', $context);
 
     if ($this->membershipID) {
-      $params = ['id' => $this->membershipID];
-      CRM_Member_BAO_Membership::retrieve($params, $values);
+      $values = \Civi\Api4\Membership::get()
```

Comment:
API calls always default to `checkPermissions = TRUE` whereas `CRM_Member_BAO_Membership::retrieve` does not check permissions. That might actually be a good change since this is a user-facing page, but it is a noteworthy change.

Reply:
Happy to take your lead on this one. Feels like checking permissions would make sense here, but equally there could be something I'm missing.

Reply:
It's weird but I don't see anywhere that ACLs are getting checked to verify access to the contact. That could be an oversight which we are correcting here.